### PR TITLE
[stable/keycloak] Update to 4.8.3 and related changes

### PR DESCRIPTION
**Pull request by thomasdarimont**
#### What this PR does / why we need it:

Updates the used Keycloak image to the latest version 4.8.3.Final.

- Added support for more reliable database connection handling.
- Referenced missing distributed caches in ha.cli configuration.
- Configured events-listener SPI to log login events to stdout for analysis

Signed-off-by: Thomas Darimont <thomas.darimont@gmail.com>

#### Special notes for your reviewer:
cc @unguiculus 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x ] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [ x] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

File: stable/keycloak/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: keycloak
-version: 4.2.1
-appVersion: 4.5.0.Final
+version: 4.3.1
```
Comment (unguiculus) [inline stable/keycloak/Chart.yaml]: Should be 4.3.0

File: stable/keycloak/README.md [hunk 2]
```diff
@@ -175,6 +175,8 @@ keycloak:
       value: DEBUG
     - name: CACHE_OWNERS:
```
Comment (unguiculus) [inline stable/keycloak/README.md]: While your at it: The colon after `CACHE_OWNERS` should not be there.

File: stable/keycloak/scripts/datasource.cli [hunk 3]
```diff
@@ -0,0 +1,12 @@
+## Query timeout in seconds
+/subsystem=datasources/data-source=KeycloakDS/:write-attribute(name=query-timeout,value=${env.DB_QUERY_TIMEOUT:300})
+
+## Postgres specific connection validation
```
Comment (unguiculus) [inline stable/keycloak/scripts/datasource.cli]: Postgres-specific stuff should only be added when Postgres is selected.

File: stable/keycloak/Chart.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
+appVersion: 4.8.2.Final
```
Comment (unguiculus) [inline stable/keycloak/Chart.yaml]: 4.8.3.Final

**Review by unguiculus — CHANGES_REQUESTED** [review]
Pleae update the Keycloak version in the readme as well.

File: stable/keycloak/Chart.yaml (hunk 1 above)
Comment (thomasdarimont) [reply stable/keycloak/Chart.yaml]: Adapted version to 4.3.0

File: stable/keycloak/scripts/datasource.cli (hunk 3 above)
Comment (thomasdarimont) [reply stable/keycloak/scripts/datasource.cli]: Added guard for postgres specific config.

File: stable/keycloak/Chart.yaml (hunk 4 above)
Comment (thomasdarimont) [reply stable/keycloak/Chart.yaml]: Adapted version to 4.8.3.Final

**Review by unguiculus — APPROVED** [review]

**Comment by alivespirit** [comment]
@thomasdarimont @unguiculus 
Datasource changes here wouldn't work, because after executing these cli scripts keycloak on startup executes `/bin/sh /opt/jboss/tools/databases/change-database.sh $DB_VENDOR` which removes KeycloakDS datasource with custom configurations applied by datasource.cli:
```
sh-4.2$ cat /opt/jboss/tools/cli/databases/postgres/change-database.cli
/subsystem=datasources/data-source=KeycloakDS: remove()
```

**Comment by thomasdarimont** [comment]
@alivespirit thanks for the hint. I think you're right. The additional config gets by the `change-database.cli` script. 
We could try to sneak the additional datasource configuration into the stock Keycloak image by appending to the particular `change-database.cli` script. However a better approach would probably be to send a PR to the jboss-dockerfiles/keycloak repository with the additional datasource configuration there.

**Comment by alivespirit** [comment]
@thomasdarimont I've tried appending custom configs to `change-database.cli` script (in `keycloak.sh`), but it won't work because `keycloak.sh` is executed from `jboss` user while `change-database.cli` is owned by `root`. So yes, best way would be to send a PR to jboss-dockerfiles repo.

**Comment by thomasdarimont** [comment]
@alivespirit for reference, the discussion on the keycloak-dev mailing-list
http://lists.jboss.org/pipermail/keycloak-dev/2019-January/011540.html

**Pull request merged**